test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs

```diff
@@ -126,23 +127,35 @@ public void Sources_WarnWhenAdding(string source, bool shouldWarn)
                     settings.ConfigPath
                 };
 
-                // Act
-                CommandRunnerResult result = _fixture.RunDotnetExpectSuccess(workingPath, string.Join(" ", args));
+                if (shouldFail)
```

Comment:
I think it would be better to split the test cases into two. My understanding is that `[InlineData]` is meant to pass multiple values for the same test case, not for having different code paths based on the value passed to the test case.